[Hub Generated] Review request for Microsoft.SqlVirtualMachine to add version preview/2021-11-01-preview

rchlkm opened the pull request:
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/rchlkm/azure-rest-api-specs/dev-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2021-11-01-preview...main/).


### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

rchlkm committed:
Adds base for updating Microsoft.SqlVirtualMachine from version preview/2017-03-01-preview to version 2021-11-01-preview

rchlkm committed:
Updates readme

rchlkm committed:
Updates API version in new specs and examples

rchlkm committed:
add 2021-11-01 swagger

rchlkm committed:
add missing files, spellcheck, add MAXDOP to custom-words

rchlkm committed:
prettier check

rchlkm committed:
fix model validation errors

rchlkm committed:
fix swagger avocado failures, SDK azure-sdk-for-net failures due to DaysOfWeek

rchlkm committed:
add system data to AvailabilityGroupListener, SqlVirtualMachineGroup, SqlVirtualMachine

j5lim reviewed (approved):
Approved from ARM side.

rchlkm committed:
update list operation for 2021-11-01

rchlkm committed:
add ms-identifier to address lintdiff R4041

rchlkm commented:
@msyyc I added the `x-ms-identifiers` property to the `AgConfiguration` object based on the R4041 error in this pipeline run: https://github.com/Azure/azure-rest-api-specs/pull/17252/checks?check_run_id=4823820712

Once I added the property, the azure-sdk-for-net pipeline run started failing here: https://github.com/Azure/azure-rest-api-specs/pull/17252/checks?check_run_id=5161611892

I believe this is because the .net sdk hasn't added support for the `x-ms-identifiers` yet. 

Can I remove the `x-ms-identifiers` property for this PR? Once .net sdk can support this property, I will raise a PR adding the value.

msyyc commented:
Hi @ArcturusZhang , please confirm breaking change of Go

ArcturusZhang commented on specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2021-11-01-preview/sqlvm.json:
```diff
@@ -0,0 +1,2243 @@
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "IdentityTypeWithNone",
```
Could we change this name back to `IdentityType`? This is a breaking change comparing with the last api-version 2017-03-01-preview. I did not see another occurrence of `IdentityType`, therefore it should be OK to make the change
```suggestion
            "name": "IdentityType",
```

rchlkm replied:
I've confirmed that we should keep it as IdentityType. Updated the PR accordingly

ArcturusZhang commented on specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/preview/2021-11-01-preview/sqlvm.json:
```diff
@@ -0,0 +1,2243 @@
+          "description": "SQL Server Log Storage Settings."
+        },
+        "sqlTempDbSettings": {
+          "$ref": "#/definitions/SQLTempDbSettings",
```
I also see this type has been changed from `SQLStorageSettings` to `SQLTempDbSettings`, **please double confirm.**
I believe this is inevitable therefore I am OK with this breaking change, but I wonder why our cross version breaking change pipeline is not reporting this.

rchlkm replied:
I've confirmed we are updating this type to `SQLTempDbSettings` as part of this api version. `SQLTempDbSettings` is an extended type of `SQLStorageSettings`

rchlkm committed:
remove x-ms-identifiers property as it is not available yet on .net sdk

rchlkm committed:
add SqlVirtualMachines_StartAssessment

rchlkm committed:
update ResourceIdentity enum from IdentityTypeWithNone to IdentityType

rchlkm committed:
update list operations json

rchlkm committed:
prettier check

rchlkm committed:
re-add ms-identifier to address lintdiff R4041

msyyc commented:
The new api-version will be deployed soon.
![image](https://user-images.githubusercontent.com/70930885/154182248-00439be0-9654-4bc7-b501-a9919ff3df2d.png)

msyyc commented:
/azp run

msyyc commented:
/azp run
